ui/dashboards/src/components/GridLayout/GridLayout.tsx

```diff
@@ -21,21 +21,26 @@ import { GridTitle } from './GridTitle';
 const ResponsiveGridLayout = WidthProvider(Responsive);
 
 export interface GridLayoutProps extends BoxProps {
+  index: number;
   definition: GridDefinition;
-  renderGridItemContent: (definition: GridItemDefinition) => React.ReactNode;
+  renderGridItemContent: (definition: GridItemDefinition, groupIndex: number) => React.ReactNode;
 }
 
 /**
  * Layout component that arranges children in a Grid based on the definition.
  */
 export function GridLayout(props: GridLayoutProps) {
   const {
+    index,
     definition: { spec },
     renderGridItemContent,
     ...others
   } = props;
 
-  const [isOpen, setIsOpen] = useState(spec.display?.collapse?.open ?? true);
+  const [isOpen, setIsOpen] = useState(!!spec.display?.collapse?.open);
+  useEffect(() => {
+    setIsOpen(!!spec.display?.collapse?.open);
+  }, [spec.display?.collapse]);
```

Comment:
Seems kinda confusing. Why do we also need this effect when we have state?

Reply:
Shan explained it nicely:

if the dashboard JSON changes, it will sync the grid collapse state. However, we also keep track at local state so it doesn't modify the original dashboard definition